### PR TITLE
Fix issue 1371: xcatprobe switch-macmap - could not get valid informa…

### PR DESCRIPTION
**Pull request by zet809**
…tion
For issue #1371 
2 more modifications:
1. create the link switchprobe if not exist whenever running './switch-macmap' 
2. deal with the error msg output by switchprobe

### CODE REVIEW

File: perl-xCAT/xCAT/MacMap.pm [hunk 1]
```diff
@@ -570,10 +570,12 @@ sub getsnmpsession {
     my $snmpver='1';
     my $swent = $self->{switchparmhash}->{$switch};
 
-  if ($swent) {
-      $snmpver=$swent->{snmpversion};
-      $community=$swent->{password};
-  }
+    if ($swent->{snmpversion}) {
```
Comment (daniceexi) [inline perl-xCAT/xCAT/MacMap.pm]: I think you should check the validity of $swent first.

Comment (zet809) [inline perl-xCAT/xCAT/MacMap.pm]: Yes, modified, pls help to review again.

**Comment by daniceexi** [comment]
I have reviewed the change and agree to merge.

File: xCAT-probe/subcmds/switch-macmap [hunk 2]
```diff
@@ -53,13 +53,15 @@ if ($help) {
     probe_utils->send_msg("$output", "d", $::USAGE);
     exit 0;
 }
+
```
Comment (hu-weihua) [inline xCAT-probe/subcmds/switch-macmap]: 1. Line 45-48 seems to check the validity of input option, I think line 36-42 can cover this. 
2. If the rules of ``noderange`` of xCAT probe commands can comply with the rules of xCAT itself, it's easier for customer to use.

File: xCAT-probe/subcmds/switch-macmap [hunk 3]
```diff
@@ -92,21 +94,25 @@ if (-f $error_file) {
 }
 my $fd;
 open($fd, "<", "$normal_file");
-my %fails = ();
+my @fails = ();
```
Comment (hu-weihua) [inline xCAT-probe/subcmds/switch-macmap]: line 78-29:  if we run ``switch-macmap`` twice at same time. that means two progress will operate ``/tmp/result_normal`` and ``/tmp/result_error``  at same time, that maybe result in unexpected problem. suggest change these two files to ``/tmp/result_normal.$$`` and ``/tmp/result_error.$$``.

**Pull request merged**